code/scripting/api/libs/ui.cpp

```diff
@@ -2479,5 +2481,309 @@ ADE_FUNC(closePause, l_UserInterface_PauseScreen, nullptr, "Makes sure everythin
+	"Mode - the mode where 0 is normal, 1 is private message, 2 is channel switch, 3 is server message, 4 is MOTD",
+	"table",
+	"A table of chat strings")
+{
+	SCP_UNUSED(L);
+
+	using namespace luacpp;
+
+	LuaTable chats = LuaTable::create(L);
+
+	for (size_t i = 0; i < Multi_pxo_chat.size(); i++) {
+		SCP_list<chat_line>::iterator line = Multi_pxo_chat.begin();
+		std::advance(line, i);
+		
+		char callsign[CALLSIGN_LEN];
+		char text[MAX_CHAT_LINE_LEN];
+
+		char* spacePos = std::strchr(line->text, ' ');
+
+		if (spacePos != nullptr) {
+			// Calculate the length of the callsign
+			int len = spacePos - line->text;
+
+			// Copy the callsign
+			std::strncpy(callsign, line->text, len);
+			callsign[len] = '\0';
+
+			// Copy the text
+			std::strcpy(text, spacePos + 1); // Copy from the character after spacePos
+		}
+
+		int mode;
+		switch (line->mode) {
+		case CHAT_MODE_PRIVATE:
+			mode = 1;
+			break;
+		case CHAT_MODE_CHANNEL_SWITCH:
+			mode = 2;
+			break;
+		case CHAT_MODE_SERVER:
+			mode = 3;
+			break;
+		case CHAT_MODE_MOTD:
+			mode = 4;
+			break;
+		default:
+			mode = 0;
+			break;
+		}
+
+		auto item = luacpp::LuaTable::create(L);
+
+		item.addValue("Callsign", luacpp::LuaValue::createValue(Script_system.GetLuaSession(), callsign));
+		item.addValue("Message", luacpp::LuaValue::createValue(Script_system.GetLuaSession(), text));
+		item.addValue("Mode", luacpp::LuaValue::createValue(Script_system.GetLuaSession(), mode));
+
+		chats.addValue(i + 1, item);
+	}
+
+	return ade_set_args(L, "t", chats);
+}
+
+ADE_FUNC(sendChat, l_UserInterface_MultiPXO, "string", "Sends a string to the current channel's IRC chat", nullptr, nullptr)
+{
+	const char* msg;
+	if (!ade_get_args(L, "s", &msg))
+		return ADE_RETURN_NIL;
+
+	multi_pxo_chat_send(msg);
+
+	return ADE_RETURN_NIL;
+}
+
+ADE_FUNC(getPlayers, l_UserInterface_MultiPXO, nullptr, "Gets the entire player list as a table of strings", "string[]", "A table of player names")
+{
+	SCP_UNUSED(L);
+
+	using namespace luacpp;
+
+	LuaTable chats = LuaTable::create(L);
+
+	for (size_t i = 0; i < Multi_pxo_players.size(); i++) {
+		chats.addValue(i + 1, Multi_pxo_players[i]);
+	}
+
+	return ade_set_args(L, "t", chats);
+}
+
+ADE_FUNC(getPlayerChannel, l_UserInterface_MultiPXO, nullptr, "Searches for a player and returns if they were found and the channel they are on. Channel is an empty string if channel is private or player is not found.", "string string", "The response string and the player's channel")
+{
+	const char* plr_name;
+	if (!ade_get_args(L, "s", &plr_name))
+		return ADE_RETURN_NIL;
+
+	SCP_string cmd;
+	sprintf(cmd, "/WHOIS %s", plr_name);
+
+	multi_pxo_chat_send(cmd.c_str());
+
+	// We have to process the pxo api in order for the response to come back
+	while (stricmp(User_req_channel, "") == 0) {
+		if (Multi_pxo_connected) {
+			multi_pxo_api_process();
+		}
+	}
+
+	SCP_string response;
+	SCP_string channel;
+
+	if ((ptr_s)User_req_channel == -1) {
```

Comment:
`User_req_channel` is never -1. That is a return value from `GetChannelByUser()` in case of error.

Reply:
No longer relevant.